sdk/keyvault/azure-security-keyvault-keys/inc/azure/keyvault/keys/deleted_key.hpp

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @brief Represents a Key Vault key that has been deleted, allowing it to be recovered, if needed.
+ *
+ */
+
+#pragma once
+
+#include <azure/core/datetime.hpp>
+
+#include "azure/keyvault/keys/key_constants.hpp"
+#include "azure/keyvault/keys/key_vault_key.hpp"
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Keys {
+
+  /**
+   * @brief Represents a Key Vault key that has been deleted, allowing it to be recovered, if
+   * needed.
+   *
+   */
+  struct DeletedKey : public KeyVaultKey
+  {
+    /**
+     * @brief A recovery url that can be used to recover it.
+     *
+     */
+    std::string RecoveryId;
+
+    /**
+     * @brief Construct an empty DeletedKey
+     *
+     */
+    DeletedKey() = default;
+
+    /**
+     * @brief Construct a new Deleted Key object.
+     *
+     * @param name THe name of the deleted key.
+     */
+    DeletedKey(std::string name) : KeyVaultKey(name) {}
+
+    /**
+     * @brief Indicate when the key was deleted.
+     *
+     */
+    Azure::Core::DateTime DeletedOn;
+
+    /**
+     * @brief Indicate when the deleted key will be purged.
+     *
+     */
+    Azure::Core::DateTime ScheduledPurgeDate;
```

Comment:
We add a postfix `Date` on one date but not the other.  Is there a design reason that we dont have Date postfix on both?

Reply:
I picked same names as .NET.
I will update to use what swagger uses:
```json
  "scheduledPurgeDate": {
                    "type": "integer",
                    "format": "unixtime",
                    "readOnly": true,
                    "description": "The time when the key is scheduled to be purged, in UTC"
                },
                "deletedDate": {
                    "type": "integer",
                    "format": "unixtime",
                    "readOnly": true,
                    "description": "The time when the key was deleted, in UTC"
                }
```